avro-cpp: builds failing with exit code 1.

### Nixpkgs version

- Unstable (26.05)

### Describe the bug

avro-cpp is failing with exit code 1.


### Steps to reproduce

```
nix build .#avro-cpp
```

### Expected behaviour

avro-cpp should be built sucessfully

### Screenshots

_No response_

### Relevant log output

```console
sudo darwin-rebuild switch --flake .#macbook                                                                                        took 3s
building the system configuration...
warning: Git tree '/Users/dinesh.reddy/repos/nixos-config' is dirty
error: Cannot build '/nix/store/32hidqiic1vz1di2xjncb0sik72sx6sn-avro-c++-1.12.0.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/8pspsf1iili7q8zqxpwlaigawhpajf8m-avro-c++-1.12.0
       Last 25 log lines:
       > CMake Warning (dev) at CMakeLists.txt:82 (find_package):
       >   Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
       >   --help-policy CMP0167" for policy details.  Use the cmake_policy command to
       >   set the policy and suppress this warning.
       >
       > This warning is for project developers.  Use -Wno-dev to suppress it.
       >
       > CMake Error at /nix/store/vxy353g9mylaw8b4fw3cs56mq5g1pn6c-boost-1.89.0-dev/lib/cmake/Boost-1.89.0/BoostConfig.cmake:141 (find_package):
       >   Could not find a package configuration file provided by "boost_system"
       >   (requested version 1.89.0) with any of the following names:
       >
       >     boost_systemConfig.cmake
       >     boost_system-config.cmake
       >
       >   Add the installation prefix of "boost_system" to CMAKE_PREFIX_PATH or set
       >   "boost_system_DIR" to a directory containing one of the above files.  If
       >   "boost_system" provides a separate development package or SDK, be sure it
       >   has been installed.
       > Call Stack (most recent call first):
       >   /nix/store/vxy353g9mylaw8b4fw3cs56mq5g1pn6c-boost-1.89.0-dev/lib/cmake/Boost-1.89.0/BoostConfig.cmake:262 (boost_find_component)
       >   /nix/store/nyj5c3998hrgrjs9ljn437kxvg9mfgsq-cmake-4.1.2/share/cmake-4.1/Modules/FindBoost.cmake:609 (find_package)
       >   CMakeLists.txt:82 (find_package)
       >
       >
       > -- Configuring incomplete, errors occurred!
       For full logs, run:
         nix log /nix/store/32hidqiic1vz1di2xjncb0sik72sx6sn-avro-c++-1.12.0.drv
error: Cannot build '/nix/store/2awx0a7q6aj2kghr0z9firrcx5mmy4c4-libserdes-8.1.0.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/r1nhslvwi3a5b77wqfnf086cz91q3mr3-libserdes-8.1.0-dev
         /nix/store/vlqbb0gsvs04gdbar2n9gpgrv8592l2h-libserdes-8.1.0
error: Cannot build '/nix/store/zg7wal1abh8162mc8pzcjag06azn2j74-kcat-1.7.1.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/8kz2c9kk5vcb2vpd7vgmn3d2gxkqk5pc-kcat-1.7.1
error: Cannot build '/nix/store/z2lj8whkal1kpjhvrifndi156s8bqa8p-home-manager-applications.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/nb52la4sjkcrvag67an3jibaca9flkwy-home-manager-applications
error: Cannot build '/nix/store/cb1y1b1b74qqwzm6zs1cvyjqn28d1nf0-home-manager-fonts.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/kr637zcvdfamh62icp6lr56v18mj4rkk-home-manager-fonts
error: Cannot build '/nix/store/zxpx6373cdga1zalhnss68k55dzc5602-home-manager-path.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/m583vn7bsrygzcwng5gip1abrdwpvgai-home-manager-path
error: Cannot build '/nix/store/ym94264zjij084f122c296fmhnrpqr16-home-manager-generation.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/bnscfcbhc0gqkzix52baczygh7qm5ccw-home-manager-generation
error: Cannot build '/nix/store/6a2z52iwi3xdav8chxqa9kd0jfrpnwlk-user-environment.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/mgggl6akwqfnbsqw6g02spxdwyzchyk8-user-environment
error: Cannot build '/nix/store/aiy92vbvxp2c90xjg3ndbmqim5wvn609-activation-dinesh.reddy.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/rnli1dkw0yr83fh13zb086d3rc2i89b3-activation-dinesh.reddy
error: Cannot build '/nix/store/94pjfkky7m0klp4mdbbxc08kddc79yn8-darwin-system-26.05.3bfa436.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/x2mq8ydn8aghrr8x80hwpv7v6f4m2xqi-darwin-system-26.05.3bfa436
```

### Additional context

Failed on Both MacOs and nixos linux system
nixpkgs version: nixpkgs-unstable

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.10, NixOS, 26.05 (Yarara), 26.05.20260217.0182a36`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.3`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/dx2qikyb4dyb6hbdfywbmsyla0z5a1h3-source`

 - system: `"aarch64-darwin"`
 - host os: `Darwin 25.3.0, macOS 26.3`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.31.3`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/q1l7xzqg7pmhylizamsdf1msv49vf1rn-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/av/avro-cpp/0004-remove-boost-system-component.patch
+++ b/pkgs/by-name/av/avro-cpp/0004-remove-boost-system-component.patch
@@ -1,0 +1,15 @@
+Remove boost_system component from find_package
+
+Boost 1.89.0 removed the separate boost_system compiled library.
+The boost::system headers are still available through the core Boost package.
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -79,7 +79,7 @@ endif ()
+
+
+ find_package (Boost 1.38 REQUIRED
+-    COMPONENTS filesystem iostreams program_options regex system)
++    COMPONENTS filesystem iostreams program_options regex)
+
+ include(FetchContent)
+ FetchContent_Declare(

--- a/pkgs/by-name/av/avro-cpp/package.nix
+++ b/pkgs/by-name/av/avro-cpp/package.nix
@@ -24,6 +24,7 @@ stdenv.mkDerivation (finalAttrs: {
     ./0001-get-rid-of-fmt-fetchcontent.patch
     ./0002-fix-fmt-name-formatter.patch
     ./0003-fix-fmt-type-formatter.patch
+    ./0004-remove-boost-system-component.patch
   ];
 
   nativeBuildInputs = [

